build: fix several compiler warnings

federico-sysdig opened the pull request:
**What type of PR is this?**
/kind cleanup

**Any specific area of the project related to this PR?**
/area build

**What this PR does / why we need it**:
Fix several warnings that come up when not all warnings are silenced.
Some of these fixes can be beneficial to provide more robust code.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1515#" title="Author self-approved">federico-sysdig</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign mstemm for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["mstemm"]} -->

incertum reviewed (commented):
ohhh that's a lot of fixes! Thank you @federico-sysdig, @LucaGuerra mind checking on them since we talked about looking into these parts of the code base more anyways, thanks!

gnosek reviewed (commented):
LGTM! I assume all the visibility changes (private<->public) are by design? :)

FedeDP commented:
/milestone 0.14.0

federico-sysdig commented:
> I assume all the visibility changes (private<->public) are by design? :)

@gnosek It's a good point. Let me list these changes and try and justify them.
- `sinsp_filter_check_container::parse_field_name`, moved from private to public as all the other 15+ instances and the base class declaration are public.
- `sinsp_filter_check_fd::extract_org`, moved from public to private as all the other 7 instances are private and, of course, used only within their classes. The same reasoning applies to `compare_ip`, `compare_net`, `compare_port`, and `compare_domain`. It could be argued that there might exist some user code that uses these formerly public functions, but they don't appear as being designed for external use. That said, I'm ready to change them back to public, if we have an idea that this is not the case.

jasondellaluce reviewed (approved):
/approve

🤩

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1515#" title="Author self-approved">federico-sysdig</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1515#pullrequestreview-1752548529" title="Approved">jasondellaluce</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [jasondellaluce]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
LGTM label has been added.  <details>Git tree hash: d8f4d4e5bd84cbefcacc71664da2c7b14180b606</details>

LucaGuerra reviewed (approved):
Thank you for this!

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1515#" title="Author self-approved">federico-sysdig</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1515#pullrequestreview-1752548529" title="Approved">jasondellaluce</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1515#pullrequestreview-1752618207" title="Approved">LucaGuerra</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [LucaGuerra,jasondellaluce]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
